compositor_pipeline/src/queue/audio_queue.rs

```diff
@@ -0,0 +1,101 @@
+use std::{
+    collections::{HashMap, VecDeque},
+    time::{Duration, Instant},
+};
+
+use super::QueueError;
+use compositor_render::{AudioSamplesBatch, AudioSamplesSet, InputId};
+
+#[derive(Debug)]
+pub struct AudioQueue {
+    /// Enqueued sample batches per output
+    input_queues: HashMap<InputId, VecDeque<AudioSamplesBatch>>,
```

Comment:
Why batching? You said you wanted to keep samples in queue?

Reply:
Keeping samples requires dealing with missing samples in the queue. As I wrote in the PR description, when I started implementing audio mixing, I noticed that the mixer already has to deal with a lot of corner cases like this and I don't think spreading this responsibility meaningful simplifies code. Mixer will have to deal with cases like:
- Different start pts of samples for different inputs (when input is added)
- Missing samples at the end (when samples haven't been received yet)
- Different input parameters, like sample rate, channels, etc.

Previously I thought that keeping just a vec of samples would simplify implementation, but now I think it just increases the number of silent implementation assumptions between mixer and queue (mixer has to know how the queue deals with missing samples, etc. to work properly). With the current approach (sending batches), the only assumption the mixer has to make about the queue is that it delivers batches with samples in the range of currently mixed fragments when possible. I think this increases testability and makes debugging easier. 

Another benefit of keeping batches is that this reduces copying and allocations, but I don't think that performance is critically important here.